### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by oxycash** [hashicorp/consul#16648]
#### Overview of the Issue

We are trying to work on #16601 and wanted to try joining server cluster using WAN port instead of LAN by disabling lan. Then the agent simply crashes.

---

```
==> Starting Consul agent...
              Version: '1.15.1'
           Build Date: '2023-03-07 20:35:33 +0000 UTC'
              Node ID: '76c16aff-1d82-3fde-0af6-113c3e458587'
            Node name: 'consul-dns'
           Datacenter: 'dc1' (Segment: '')
               Server: false (Bootstrap: false)
          Client Addr: [127.0.0.1] (HTTP: 92208, HTTPS: -1, gRPC: 92209, gRPC-TLS: -1, DNS: 92210)
         Cluster Addr: 10.144.8.126 (LAN: -1, WAN: 92207)
    Gossip Encryption: true
     Auto-Encrypt-TLS: false
            HTTPS TLS: Verify Incoming: false, Verify Outgoing: false, Min Version: TLSv1_2
             gRPC TLS: Verify Incoming: false, Min Version: TLSv1_2
     Internal RPC TLS: Verify Incoming: false, Verify Outgoing: false (Verify Hostname: false), Min Version: TLSv1_2

==> Log data will now stream in as it occurs:

2023-03-16T09:33:46.784Z [WARN]  agent: using enable-script-checks without ACLs and without allow_write_http_from is DANGEROUS, use enable-local-script-checks instead, see https://www.hashicorp.com/blog/protecting-consul-from-rce-risk-in-specific-configurations/
2023-03-16T09:33:46.790Z [WARN]  agent.auto_config: using enable-script-checks without ACLs and without allow_write_http_from is DANGEROUS, use enable-local-script-checks instead, see https://www.hashicorp.com/blog/protecting-consul-from-rce-risk-in-specific-configurations/
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2a02c2a]

goroutine 1 [running]:
github.com/hashicorp/consul/agent.newConsulConfig(0xc000e0d000, {0x43152a0, 0xc000ced2c0})
        github.com/hashicorp/consul/agent/agent.go:1291 +0x1ea
github.com/hashicorp/consul/agent.(*Agent).Start(0xc000cdd500, {0x4300260, 0xc0000d0730})
        github.com/hashicorp/consul/agent/agent.go:579 +0x43d
github.com/hashicorp/consul/command/agent.(*cmd).run(0xc00013e000, {0xc0000720e0, 0x2, 0x2})
        github.com/hashicorp/consul/command/agent/agent.go:225 +0x112f
github.com/hashicorp/consul/command/agent.(*cmd).Run(0xc00013e000, {0xc0000720e0?, 0xffffffffffffffff?, 0x0?})
        github.com/hashicorp/consul/command/agent/agent.go:70 +0x27
github.com/mitchellh/cli.(*CLI).Run(0xc000993b80)
        github.com/mitchellh/cli@v1.1.0/cli.go:260 +0x5f8
main.realMain()
        github.com/hashicorp/consul/main.go:48 +0x40f
main.main()
        github.com/hashicorp/consul/main.go:18 +0x19
```

**Comment by huikang** [comment]
I don't think the serf lan port can be disabled (there is a recent fix #16715 to disallowing disabling serf lan port). Could you provide more context of your use case?

**Comment by david-yu** [comment]
Will close due to no response. Please file a new issue if needed.
